### DISCUSSION TITLE
Intercepting and Parallel routing

### DISCUSSION THREAD

**Discussion by jekingohel** [vercel/next.js#47270] [Help]
### Summary

Hi,

I have read blog where I can create intercepting and parallel routing. but neither I could not find complete documentation nor find any example. would be good if any one can help to understand this concept.

Blog link:
https://nextjs.org/blog/layouts-rfc#advanced-routing-patterns

I want to build modal based routing.
- Modals being accessible through an URL.
- Navigating backwards and forwards using client-side navigation should close and reopen the modal.
- Main page will be display if modal is opened (both in refresh and navigation)
 for example, [username]/xyz content will be display when I access modal routing.

### Example

https://nextjs.org/blog/layouts-rfc#combining-intercepting-and-parallel-routes

**Comment by tommyboylab** [comment]
The latest canary has tests for it here: https://github.com/vercel/next.js/tree/canary/test/e2e/app-dir/parallel-routes-and-interception/app/intercepting-parallel-modal

Maybe you could use it as a guide?
